package.json

```diff
@@ -52,6 +52,7 @@
     "react": "17.0.1",
     "react-intl": "5.24.6",
     "react-native": "0.64.3",
+    "react-native-codegen": "^0.0.7",
```

Comment:
The RN repo went back and forth between having and not having this
    dependency in the template app, and I found it hard to follow why or
    what it does. But the RN v0.65 release was made with the dep
    included at this version range, so, might as well?

Following your handy links to those commits, it looks like when it got removed from the template dependencies, it was because it got moved inside the `react-native` NPM package itself.

Thinking about that, I think it means the ideal thing is probably to hold back this change to do in the same commit with the RN upgrade itself. At our current version, this isn't in the (template) package.json because it's closely coupled with the code in react-native itself.

I'm not sure whether adding it to our package.json here would do nothing (and RN would keep on invoking its own internal version of react-native-codegen), or would result in using the new react-native-codegen version from NPM. If the former, then adding it here is potentially confusing because it looks like it does something but doesn't actually. If the latter, then in principle it's incompatible -- our current RN version was explicitly meant to use its internal copy of react-native-codegen, not a separately-released one from NPM.

Reply:
Cool, yeah, makes sense.

> Following your handy links to those commits

For posterity, those commits were:

  facebook/react-native@58c80d4f8
  facebook/react-native@e99b8bbb4
  facebook/react-native@cd6c9f327
  facebook/react-native@98e173445
  facebook/react-native@e77595784

I think I'll make the main upgrade commit less verbose by omitting that list and linking here.